### PR TITLE
#55 Add a Compressed Json Serializer using the GZipStream

### PR DESCRIPTION
**Pull request by fgideali**
PR for #55

### CODE REVIEW

File: src/CacheManager.Serialization.Json/project.json [hunk 1]
```diff
@@ -33,9 +33,17 @@
     "Newtonsoft.Json": "6.0.8"
   },
   "frameworks": {
-    "dnx451": { },
+    "dnx451": {
+      "dependencies": {
+        "System.IO.Compression": "4.1.0-beta-23516"
```
Comment (MichaCo) [inline src/CacheManager.Serialization.Json/project.json]: I think for dnx451 and net45, you should not add .net core assemblies. Either the Compression stuff is already available, or you add them to `frameworkAssemblies`.

File: src/CacheManager.Serialization.Json/GzJsonCacheSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
+﻿#if !DOTNET5_2
```
Comment (MichaCo) [inline src/CacheManager.Serialization.Json/GzJsonCacheSerializer.cs]: Is there any particular reason you exclude .net core? The functionality should be available, even for DOTNET5.2 I think

**Comment by MichaCo** [comment]
Thanks for the PR @fgideali, looks good overall. Don't worry about the 2nd check which is failing, I'll take care of all that after a merge.

Could you take a look at the project.json references? I think it is wrong how it is setup right now.
And targeting .net core should work, too, I think

**Comment by fgideali** [comment]
Indeed, the System.IO.Compression dependency is only needed for the .net core
I fixed the project references.

Is there anything else I could do to prevent that fail on AppVeyor?

**Comment by MichaCo** [comment]
Thanks!
For the 2nd solution to build, the file needs to be added as reference to the old csproj. 
But don't worry, I'll go through that when I merge the changes and create a new release. 
Hopefully I find some time this weekend... ;)

**Pull request merged**